src/macros.js

```diff
@@ -235,8 +263,8 @@ defineMacro("\\thickspace", "\\;");   //   \let\thickspace\;
 
 // \DeclareRobustCommand\hspace{\@ifstar\@hspacer\@hspace}
 // \def\@hspace#1{\hskip  #1\relax}
-// KaTeX doesn't do line breaks, so \hspace is the same as \kern
-defineMacro("\\hspace", "\\kern{#1}");
+// KaTeX doesn't do line breaks, so \hspace and \hspace* are the same as \kern
+defineMacro("\\hspace", "\\@ifstar\\kern\\kern");
```

Comment:
I would've assumed that `\hspace*` would be aliased to `\hspace` by doing:
```
defineMacro("\\hspace*", "\\hspace")
```
My hope was to one day we might have a script that scrapes the source code to determine all of the commands we support.  This makes it harder.  If this is the way LaTeX defines `*` functions then let's roll with it.

Reply:
If a person wrote:
```
defineMacro("\\hspace", "\\@ifstar{\\kern{#1}\\kern{#1}}\\kern{#1}");
```
would that result in `\hspace*` producing double the amount of kerning?

Reply:
Yes, this is how LaTeX does it.  `\hspace*` is parsed by TeX as two tokens, a control word `\hspace` and a character `*`, and the former detects the latter and adapts its behavior.  Of course, LaTeX also does this (at least internally) to detect optional arguments, and KaTeX deals with optional arguments to functions in a different way, so this is open to debate...  But I don't think LaTeX offers another way to detect asterisks, while it does offer another way to parse optional arguments.

Anyway, `defineMacro("\\hspace*", "\\hspace")` would be difficult to get working, as it leads to ambiguous parsing.  When KaTeX sees an `\hspace` token, it doesn't know whether `*` is an argument to `\hspace` or it's just the beginning of `\hspace*`.

Regarding your second question:
```latex
defineMacro("\\hspace", "\\@ifstar{\\kern{#1}\\kern{#1}}{\\kern{#1}}");
```
should behave as you describe.  (Needed some extra braces, I think.)

Reply:
Good point about the ambiguity of the parsing.